bouncer/tests/btc_vault_swap.ts

```diff
@@ -12,30 +12,51 @@ import {
 } from '../shared/utils';
 import { getChainflipApi, observeEvent } from '../shared/utils/substrate';
 import { getBalance } from '../shared/get_balance';
-import { jsonRpc } from '../shared/json_rpc';
+import { brokerApiRpc } from '../shared/json_rpc';
+import { getEarnedBrokerFees } from './broker_fee_collection';
 
 /* eslint-disable @typescript-eslint/no-use-before-define */
-export const testBtcVaultSwap = new ExecutableTest('Btc-Vault-Swap', main, 100);
+export const testBtcVaultSwap = new ExecutableTest('Btc-Vault-Swap', main, 120);
+
+// Fee to use for the broker and affiliates
+const commissionBps = 100;
 
 interface VaultSwapDetails {
   chain: string;
   nulldata_utxo: string;
   deposit_address: string;
 }
 
+interface Beneficiary {
+  account: string;
+  bps: number;
+}
+
 async function buildAndSendBtcVaultSwap(
   depositAmountBtc: number,
   brokerUri: string,
   destinationAsset: Asset,
   destinationAddress: string,
   refundAddress: string,
+  affiliateUri1: string | null = null,
+  affiliateUri2: string | null = null,
 ) {
   await using chainflip = await getChainflipApi();
 
   const broker = createStateChainKeypair(brokerUri);
   testBtcVaultSwap.debugLog('Broker:', broker.address);
   testBtcVaultSwap.debugLog(`Btc endpoint is set to`, BTC_ENDPOINT);
 
+  const affiliates: Beneficiary[] = [];
+  if (affiliateUri1) {
+    const affiliate1 = createStateChainKeypair(affiliateUri1);
+    affiliates.push({ account: affiliate1.address, bps: commissionBps });
+  }
+  if (affiliateUri2) {
+    const affiliate1 = createStateChainKeypair(affiliateUri2);
```

Comment:
```suggestion
    const affiliate2 = createStateChainKeypair(affiliateUri2);
```

Alghough maybe instead of passing each affiliate as an individual parameter, you could pass an array instead?